File: xybanner/src/main/java/com/xyuan/xybanner/transformer/AccordionTransformer.java
package com.xyuan.xybanner.transformer;

import android.view.View;

/**
 * 手风琴折叠样式
 */
public class AccordionTransformer extends ABaseTransformer {

    @Override
    protected void onTransform(View view, float position) {
        view.setPivotX(position < 0 ? 0 : view.getWidth());
        view.setScaleX(position < 0 ? 1f + position : 1f - position);
    }

}

File: xybanner/src/main/java/com/xyuan/xybanner/transformer/AccordionTransformer.kt
package com.xyuan.xybanner.transformer

import android.view.View

/**
 * 手风琴折叠样式
 */
class AccordionTransformer : ABaseTransformer() {

    override fun onTransform(view: View, position: Float) {
        view.pivotX = (if (position < 0) 0 else view.width).toFloat()
        view.scaleX = if (position < 0) 1f + position else 1f - position
    }

}
